src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -273,68 +229,143 @@ public void close() throws IOException {
+        connectorConfig.kerberosTicketRenewPeriodMs()
+    );
+    ticketRenewThread.start();
+  }
 
-      topicPartitionWriters = new HashMap<>();
-      for (TopicPartition tp : context.assignment()) {
-        TopicPartitionWriter topicPartitionWriter = new TopicPartitionWriter(
-            tp,
-            storage,
-            writerProvider,
-            newWriterProvider,
-            partitioner,
-            config,
-            context,
-            avroData,
-            hiveMetaStore,
-            hive,
-            schemaFileReader,
-            executorService,
-            hiveUpdateFutures,
-            time
-        );
-        topicPartitionWriters.put(tp, topicPartitionWriter);
-      }
-    } catch (ClassNotFoundException
-            | IllegalAccessException
-            | InstantiationException
-            | InvocationTargetException
-            | NoSuchMethodException e
-    ) {
-      throw new ConnectException("Reflection exception: ", e);
-    } catch (IOException e) {
-      throw new ConnectException(e);
+  private void initializeHiveServices(Configuration hadoopConfiguration) {
+    hiveDatabase = connectorConfig.getString(HiveConfig.HIVE_DATABASE_CONFIG);
+    hiveMetaStore = new HiveMetaStore(hadoopConfiguration, connectorConfig);
+    if (format != null) {
+      hive = format.getHiveUtil(connectorConfig, hiveMetaStore);
+    } else if (newFormat != null) {
+      final io.confluent.connect.storage.hive.HiveUtil newHiveUtil
+          = ((HiveFactory) newFormat.getHiveFactory())
+          .createHiveUtil(connectorConfig, hiveMetaStore);
+      hive = new HiveUtil(connectorConfig, hiveMetaStore) {
+        @Override
+        public void createTable(
+            String database, String tableName, Schema schema,
+            Partitioner partitioner
+        ) {
+          newHiveUtil.createTable(database, tableName, schema, partitioner);
+          log.debug("Created Hive table {}", tableName);
+        }
+
+        @Override
+        public void alterSchema(String database, String tableName, Schema schema) {
+          newHiveUtil.alterSchema(database, tableName, schema);
+          log.debug("Altered Hive table {}", tableName);
+        }
+      };
+    } else {
+      throw new ConnectException("One of old or new format classes must be provided");
+    }
+    executorService = Executors.newSingleThreadExecutor();
+    hiveUpdateFutures = new LinkedList<>();
+  }
+
+  private void initializeTopicPartitionWriters(Set<TopicPartition> assignment) {
+    for (TopicPartition tp : assignment) {
+      TopicPartitionWriter topicPartitionWriter = new TopicPartitionWriter(
+          tp,
+          storage,
+          writerProvider,
+          newWriterProvider,
+          partitioner,
+          connectorConfig,
+          context,
+          avroData,
+          hiveMetaStore,
+          hive,
+          schemaFileReader,
+          executorService,
+          hiveUpdateFutures,
+          time
```

Comment:
🤮

Reply:
agree with that sentiment of how this is done currently, I aimed to scope this PR on purpose and can follow up on cleaning up this part